guardrails/validator_base.py

```diff
@@ -167,8 +171,36 @@ class Validator:
     _metadata = {}
 
     def __init__(
-        self, on_fail: Optional[Union[Callable, OnFailAction]] = None, **kwargs
+        self,
+        use_local: bool = None,
+        validation_endpoint: str = None,
+        on_fail: Optional[Union[Callable, OnFailAction]] = None,
```

Comment:
If we're going to be adding non-callables to this I feel like it might make sense to make this a regular class instead of a dataclass.  If this remains a dataclass there might be initializer tricks that skip calling __init__, so we should move the above defaults into the __init__ and drop the dataclass annotation.  Or we can go the other way and make these defaults in the dataclass.  I'm not as enthusiastic about the mix of data/non-data initializers.  No idea how this will behave in practice.

Reply:
Yes, I didn't even notice this had been changed to a dataclass. Dataclasses usually only have a post_init that runs, Is the dataclass __init__ being overriden for any particular reason?

Reply:
I think making this a dataclass was a hack for pydantic 1.x back in the day.  We can/should try removing the annotation and see how the QA checks play out, but I'd rather not mess with it as part of this PR since it's unrelated.

Reply:
added a TODO